ironfish-rust/src/proofs/circuit/mint_asset.rs

```diff
@@ -171,29 +136,14 @@ impl Circuit<bls12_381::Scalar> for MintAsset {
             .asset_info
             .as_ref()
             .and_then(|ai| Some(ai.asset_type()));
-        // Witness the asset type
-        // TODO: Does this properly verify that the spend note is the right asset generator?
-        // Could this be spoofed, or does this not matter? Need to consider what's public/private
-        // In other words: Does this verify that the actual note's asset generator is valid
-        // Or are we allowing this to be _any_ generator
-        let asset_generator = ecc::EdwardsPoint::witness(
-            cs.namespace(|| "asset_generator"),
-            asset_type
-                .as_ref()
-                .and_then(|at| at.asset_generator().into()),
-        )?;
 
-        let value_commitment_generator = ecc::EdwardsPoint::witness(
-            cs.namespace(|| "value commitment generator"),
-            asset_type
+        let public_address_bits = slice_into_boolean_vec_le(
```

Comment:
One downside here is that we're duplicating work which means more constraints and a slower circuit - I guess we'll have to find the right balance. I was thinking about adding an extra param to the preimage helper fn like `(..., public_address_bits: Option<Vec<Boolean>>)` but that makes things quite a bit more complex. We will probably need to think about this a bit more

Reply:
Yeah, I added a TODO for now. I think we'll revisit this with upcoming refactor so I'll defer to then.